msal_extensions/persistence.py

```diff
@@ -87,11 +106,32 @@ def save(self, content):
     def load(self):
         # type: () -> str
         """Load content from this persistence"""
-        with open(self._location, 'r') as handle:
-            return handle.read()
+        try:
+            with open(self._location, 'r') as handle:
+                return handle.read()
+        except EnvironmentError as exp:  # EnvironmentError in Py 2.7 works across platform
+            if exp.errno == errno.ENOENT:
+                raise PersistenceNotFound(
+                    message=(
+                        "Persistence not initialized. "
+                        "You can recover by calling a save() first."),
+                    where=self._location,
+                    )
+            raise
+
 
     def time_last_modified(self):
-        return os.path.getmtime(self._location)
+        try:
+            return os.path.getmtime(self._location)
+        except EnvironmentError as exp:  # EnvironmentError in Py 2.7 works across platform
+            if exp.errno == errno.ENOENT:
+                raise PersistenceNotFound(
```

Comment:
One more thing. Now that we start to work on this PR, @abhidnya13 , would you mind backport your test cases from #65 to this PR?

Reply:
@rayluo You can go ahead and backport the test cases, I think they still might need some changes and cannot be used as is. Right now it the tests are of the token cache layer

Reply:
Test cases added. But automation only covers few platforms. Please help manually test the Windows platform, and optionally the mac. I'll manually test on Linux.

Reply:
Manually tested the windows platform and mac and the tests for those are green so looks good !

Reply:
Manual test on Linux also passed.

```
(env37) demo@antix1:~/Live-usb-storage/repo/microsoft-authentication-extensions-for-python (handle-error-of-load-before-save)$ pytest tests/test_persistence.py -v                                                                                                             
============================================================================================================================= test session starts ==============================================================================================================================
platform linux -- Python 3.7.3, pytest-5.2.1, py-1.8.0, pluggy-0.13.0 -- /home/demo/Live-usb-storage/env37/bin/python3
cachedir: .pytest_cache
rootdir: /home/demo/Live-usb-storage/repo/microsoft-authentication-extensions-for-python
collected 8 items                                                                                                                                                                                                                                                              

tests/test_persistence.py::test_file_persistence PASSED                                                                                                                                                                                                                  [ 12%]
tests/test_persistence.py::test_nonexistent_file_persistence PASSED                                                                                                                                                                                                      [ 25%]
tests/test_persistence.py::test_file_persistence_with_data_protection SKIPPED                                                                                                                                                                                            [ 37%]
tests/test_persistence.py::test_nonexistent_file_persistence_with_data_protection SKIPPED                                                                                                                                                                                [ 50%]
tests/test_persistence.py::test_keychain_persistence SKIPPED                                                                                                                                                                                                             [ 62%]
tests/test_persistence.py::test_nonexistent_keychain_persistence SKIPPED                                                                                                                                                                                                 [ 75%]
tests/test_persistence.py::test_libsecret_persistence PASSED                                                                                                                                                                                                             [ 87%]
tests/test_persistence.py::test_nonexistent_libsecret_persistence PASSED  <=== Manual test passed
```